moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -0,0 +1,216 @@
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik LLC nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Henning Kayser */
+
+#pragma once
+
+#include <moveit/moveit_cpp/moveit_cpp.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/planning_pipeline/planning_pipeline.h>
+#include <moveit/robot_state/robot_state.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <moveit/robot_state/conversions.h>
+#include <moveit_msgs/MoveItErrorCodes.h>
+
+namespace moveit
+{
+namespace planning_interface
+{
+MOVEIT_CLASS_FORWARD(PlanningComponent);
+
+class PlanningComponent
+{
+public:
+  MOVEIT_STRUCT_FORWARD(PlanSolution);
+
+  class MoveItErrorCode : public moveit_msgs::MoveItErrorCodes
+  {
+  public:
+    MoveItErrorCode()
+    {
+      val = 0;
+    }
+    MoveItErrorCode(int code)
+    {
+      val = code;
+    }
+    MoveItErrorCode(const moveit_msgs::MoveItErrorCodes& code)
+    {
+      val = code.val;
+    }
+    explicit operator bool() const
+    {
+      return val == moveit_msgs::MoveItErrorCodes::SUCCESS;
+    }
+    bool operator==(const int c) const
+    {
+      return val == c;
+    }
+    bool operator!=(const int c) const
+    {
+      return val != c;
+    }
+  };
+
+  /// The representation of a plan solution
+  struct PlanSolution
+  {
+    /// The full starting state used for planning
+    moveit_msgs::RobotState start_state;
+
+    /// The trajectory of the robot (may not contain joints that are the same as for the start_state_)
+    robot_trajectory::RobotTrajectoryPtr trajectory;
+
+    /// Error code
+    MoveItErrorCode error_code;
+
+    explicit operator bool() const
+    {
+      return bool(error_code);
+    }
+  };
+
+  /// Planner parameters provided with the MotionPlanRequest
+  struct PlanRequestParameters
```

Comment:
Why is the PlannerOptions struct so similar to this struct? Do we really need both? They have the same contents. Let's remove one.